src/reformat.c

```diff
@@ -457,6 +457,12 @@ static avifResult avifImageYUVAnyToRGBAnySlow(const avifImage * image,
         uint8_t * ptrG = &rgb->pixels[state->rgbOffsetBytesG + (j * rgb->rowBytes)];
         uint8_t * ptrB = &rgb->pixels[state->rgbOffsetBytesB + (j * rgb->rowBytes)];
 
+        // If toAlphaMode is active (not no-op), assert that the appropriate alpha plane pointer is
+        // present (for this image's depth). The end of the avifPrepareReformatState() function should
+        // ensure this, but this assert makes it clear to clang's analyzer.
+        assert((state->toRGBAlphaMode == AVIF_ALPHA_MULTIPLY_MODE_NO_OP) ||
+               (((image->depth == 8) && ptrA8) || ((image->depth > 8) && ptrA16)));
+
```

Comment:
An alternative is to move this assertion to the code that uses `ptrA8` and `ptrA16`, inside the if statement that starts at line 634:

```
            if (state->toRGBAlphaMode != AVIF_ALPHA_MULTIPLY_MODE_NO_OP) {
                // Calculate A
                uint16_t unormA;
                if (image->depth == 8) {
                    unormA = ptrA8[i];
                } else {
                    unormA = AVIF_MIN(ptrA16[i], yuvMaxChannel);
                }
                ...
```

Do you prefer the current location of the assertion because we assert immediately after initializing `ptrA8` and `ptrA16`?